bitwig-studio2: 2.3.5 -> 2.4.3

gnidorah opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

FRidh commented:
I recommend using `(auto)patchelf` instead of using `LD_LIBRARY_PATH`. Note that studio1 does both.

gnidorah commented:
@FRidh Interesting. I see that bitwig-studio2 also adds `ffmpeg` to `binPath`, but that doesn't get added in resulting wrapper.

Commit:
bitwig-studio: switch to autoPatchelfHook

gnidorah commented:
@FRidh How about that? Not yet tested thoroughly though.

FRidh commented:
Check with the maintainers of the expressions.

gnidorah commented:
@michalrus @mrVanDalo What do you think?

Commit:
bitwig-studio2: 2.3.5 -> 2.4.3

mrVanDalo commented:
@gnidorah did you check  bitwig1 as well, I can check it today. I usually check if playback, save, load and keyboards shortcuts work.

mrVanDalo commented:
@gnidorah I tested bitwig-studio1, save and open does not work anymore:
```
com.bitwig.base.io.ProcessException: Error executing command zenity --file-selection --title "Save Project" --save --confirm-overwrite --file-filter "Bitwig Project | *.bwproject | *.bwsong | *.fls" --filename "/home/palo/Bitwig Studio/Projects/": Cannot run program "zenity": error=2, No such file or directory
```

I guess zenity is missing.

I tested it using `nix-review pr 53451` and `./result/bitwig-studio1/bin/bitwig-studio`

gnidorah commented:
@mrVanDalo Thanks! I've only tested VSTs, controller and tracks switching :frowning: You're correct that I've broke file dialogs. `runtimeDependencies` add dependencies to RPATH, while we need them in PATH, so I've reverted that change.
In addition, I see that bitwig-studio1 misses ffmpeg in PATH, so its impossible to drag&drop mp3 files from files panel onto audio track. Added it, now that function works (tracks playback also works).

mrVanDalo commented:
@gnidorah I checked your changes for bitwig-studio1 and everything seem fine. Again couldn't test bitwig-studio2. For me it's a go.

gnidorah commented:
@christoph00 could you please test and confirm that this pr works for you? You could use `nix-review pr 53451` command for that

christoph00 commented:
> @christoph00 could you please test and confirm that this pr works for you? You could use `nix-review pr 53451` command for that

Open/Save Dialogs do not work.

Mic92 commented:
@gnidorah @christoph00 cannot open file dialogs because bitwig is missing the `wrapGAppsHook`, which would add desktop schema files and other modules.
This could look like this: ac8acb283c6380d01e6a9d6722ccebb4cdddaa9c
Also you don't need `glib` and `gnome3.gsettings_desktop_schemas` when you already have gtk in your `buildInputs`.

Commit:
bitwig-studio: support native filesystem dialogs

gnidorah commented:
@Mic92 Thanks! This way? I have no such issue, so can't test whether my last change fixed it.

christoph00 commented:
I have still no Dialogs. Do I need some gtk stuff? Im using plasma5

Mic92 committed:
bitwig-studio: gappsWrapperArgs must read in postFixup

They are not yet set in the installPhase.

Mic92 commented:
The gappsWrapperArgs are only available in the fixup phase. I fixed that
and now it adds the desktop schemacs correctly.
I cannot test this because I don't have a license, so please @christoph00 test again.

Mic92 commented on pkgs/applications/audio/bitwig-studio/bitwig-studio1.nix:
```diff
@@ -67,22 +76,11 @@ stdenv.mkDerivation rec {
         --prefix PATH : "${binPath}" \
-        --prefix LD_LIBRARY_PATH : "${libPath}" \
+        "''${gappsWrapperArgs[@]}" \
         --set LD_PRELOAD "${libxkbcommon.out}/lib/libxkbcommon.so" || true
```
It might be a bit cleaner though to just extend `gappsWrapperArgs` and remove `dontWrapGApps`

```
gappsWrapperArgs+=("--prefix" "PATH" ":" "${binPath}")
gappsWrapperArgs+=("--set" "LD_PRELOAD" "${libxkbcommon.out}/lib/libxkbcommon.so")
```

gnidorah commented:
@Mic92 thank you!
